libs/ui/src/admin/CreateCollectionModal/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,236 @@
+import { useCreateCollection } from '@wsvvrijheid/utils'
+import { useForm } from 'react-hook-form'
+import { FaCheck, FaTimes, FaPlusCircle } from 'react-icons/fa'
+import * as yup from 'yup'
+
+import { FormItem, Navigate } from '../../components'
+import { FileUploader } from '../../components/FileUploader'
+import { CollectionCreateSuccessAlert } from './CreateCollectionSuccessAlert'
+import {
+  CreateCollectionFormFieldValues,
+  CreateCollectionModalProps,
+} from './types'
+
+const schema = () =>
+  yup.object({
+    title: yup.string().required('Title is required'),
+    description: yup.string().required('Description is required'),
+  })
+
+// TODO Consider adding modal form instead of a new page
+export const CreateCollectionModal: FC<CreateCollectionModalProps> = ({
+  auth,
+  queryKey,
+}) => {
+  const [images, setImages] = useState<Blob[]>([])
+  const [previews, setPreviews] = useState<string[]>([])
+
+  const cancelRef = useRef<HTMLButtonElement>(null)
+  const formDisclosure = useDisclosure()
+  const successDisclosure = useDisclosure()
+  const toast = useToast()
+
+  const {
+    register,
+    formState: { errors, isValid },
+    handleSubmit,
+    reset: resetForm,
+  } = useForm<CreateCollectionFormFieldValues>({
+    resolver: yupResolver(schema()),
+    mode: 'all',
+  })
+
+  useEffect(
+    () => () => {
+      // Make sure to revoke the data uris to avoid memory leaks
+      images.forEach(image => URL.revokeObjectURL((image as any).preview))
+    },
+    [images],
+  )
+
+  const { mutate, isLoading } = useCreateCollection(queryKey)
+
+  const createCollection = async (
+    data: CreateCollectionFormFieldValues & { image: Blob },
+  ) => {
+    if (!auth.user) return
+
+    const slug = slugify(data.title)
+    const formBody: CollectionCreateInput = {
+      ...data,
+      slug,
+      locale: 'tr',
+    }
+
+    mutate(formBody, {
+      onSuccess: () => {
+        formDisclosure.onClose()
+        successDisclosure.onOpen()
+      },
+      onError: () => {
+        toast({
+          title: 'Error',
+          description: 'Something went wrong',
+          status: 'error',
+          duration: 5000,
+          isClosable: true,
+        })
+      },
+    })
+  }
+
+  const handleCreateCollection = async (
+    data: CreateCollectionFormFieldValues,
+  ) => {
+    createCollection({ ...data, image: images[0] })
+  }
+
+  const resetFileUploader = () => {
+    setImages([])
+    setPreviews([])
+  }
+
+  const closeForm = () => {
+    resetFileUploader()
+    resetForm()
+    formDisclosure.onClose()
+  }
+
+  return (
+    <>
+      {/* SUCCESS ALERT */}
+      <CollectionCreateSuccessAlert
+        isOpen={successDisclosure.isOpen}
+        onClose={successDisclosure.onClose}
+        ref={cancelRef}
+      />
+
+      <Button size="lg" colorScheme="blue" onClick={formDisclosure.onOpen}>
+        <Box mr={{ base: 0, lg: 4 }}>
+          <FaPlusCircle />

```

Comment:
We should have updated the design. I prefer using `FaPlus` as icon.

```suggestion
          <FaPlus />
```